backend/env/env.go

```diff
@@ -11,20 +11,43 @@ import (
 // NewBackend creates a configuration loader that loads from the environment.
 // If the key is not found, this backend tries again by turning any kebabcase key to snakecase and
 // lowercase letters to uppercase.
-func NewBackend() backend.Backend {
+func NewBackend(fns ...opt) backend.Backend {
 	return backend.Func("env", func(ctx context.Context, key string) ([]byte, error) {
-		val, ok := os.LookupEnv(key)
+		key = strings.Replace(key, "-", "_", -1)
```

Comment:
This is a backwardly incompatible change. The original logic would succeed if an environment variable exists with exactly the same name, but it won't any longer (I'm surprised the tests pass actually - we should fix that).